### ISSUE TITLE
Introduce profile-based override for application.properties

### ISSUE THREAD

**Issue by Edubits** [quarkusio/quarkus#1615]
In Thorntail this was handled in a nice way: https://docs.thorntail.io/2.3.0.Final/#default-thorntail-yaml-files_thorntail
It would be very useful if Quarkus would support something similar to prevent using several -D properties for overriding properties (like datasource) for different staging environments or local deployments.

**Comment by gsmet** [comment]
@Edubits I confirm this is planned for a future release.

**Comment by machi1990** [comment]
@dmlloyd @gsmet what's the status of this? 

If no one is not working on  it yet, I can give it a go: basing my work on a design decision if it exists or the help of a mentor.

**Comment by dmlloyd** [comment]
It has to come after #284 which also adds support for command-line arguments and configuration changes to support them, which in turn can be used to select a profile.  We do have code to support profiles with SmallRye Config already though, so the bulk of that work is done.  I'm working on #284 now; I'd say I'm about halfway done.  Maybe another week or two.

**Comment by machi1990** [comment]
@dmlloyd thanks for clarification. Let me look on another issue.
